pkgs/development/python-modules/hacking/default.nix

```diff
@@ -7,29 +7,32 @@
   stestr,
   ddt,
   testscenarios,
+  setuptools,
+  pythonOlder,
 }:
 
 buildPythonPackage rec {
   pname = "hacking";
-  version = "6.1.0";
-  format = "setuptools";
+  version = "7.0.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-3lBqMSQDThi046acld5JjDRgvLxJwWQ9MXjRW8barBQ=";
+    hash = "sha256-ubbC5SgPfVT6gsWP4JmD9oxbb2NKw/ozn4uhalcVyrc=";
   };
 
   postPatch = ''
     sed -i 's/flake8.*/flake8/' requirements.txt
-    substituteInPlace hacking/checks/python23.py \
-      --replace 'H236: class Foo(object):\n    __metaclass__ = \' 'Okay: class Foo(object):\n    __metaclass__ = \'
-    substituteInPlace hacking/checks/except_checks.py \
-      --replace 'H201: except:' 'Okay: except:'
   '';
```

Comment:
We can probably drop postPatch now entirely.

Reply:
Hello @SuperSandro2000, thanks for the review!

Unfortunately we cannot. Because if I remove this substitution and replace it by pythonRelaxDeps, stestr will fail because a test in hacking is [trying to parse the requirements.txt file](https://github.com/openstack/hacking/blob/ad22d7ce9614ad6eb6715e35c2111874b615bba1/hacking/tests/test_doctest.py#L74C32-L74C49).

And it will result with a `pkg_resources.VersionConflict` error.

<details>

```
=========================
Failures during discovery
=========================
--- import errors ---
Failed to call load_tests:
Traceback (most recent call last):
  File "/nix/store/53kwxjyrlpj47nhv6w8hv4kaa8x6rmka-python3-3.11.9/lib/python3.11/unittest/loader.py", line 138, in loadTestsFromModule
    return load_tests(self, tests, pattern)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/hacking-7.0.0/hacking/tests/test_doctest.py", line 78, in load_tests
    check = entry.load()
            ^^^^^^^^^^^^
  File "/nix/store/jdylj8vphfpfcc11p1aj3i9912jm7mgb-python3.11-setuptools-72.1.0/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2770, in load
    self.require(*args, **kwargs)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/jdylj8vphfpfcc11p1aj3i9912jm7mgb-python3.11-setuptools-72.1.0/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2798, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/jdylj8vphfpfcc11p1aj3i9912jm7mgb-python3.11-setuptools-72.1.0/lib/python3.11/site-packages/pkg_resources/__init__.py", line 896, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/nix/store/jdylj8vphfpfcc11p1aj3i9912jm7mgb-python3.11-setuptools-72.1.0/lib/python3.11/site-packages/pkg_resources/__init__.py", line 942, in _resolve_dist
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (flake8 7.1.0 (/nix/store/3n40f271sb4ws37sc1llwkj6f7gxzk5l-python3.11-flake8-7.1.0/lib/python3.11/site-packages), Requirement.parse('flake8~=7.1.1'))
```

</details>